src/main/lrsql/util/auth.clj

```diff
@@ -132,25 +133,33 @@
            _auth
            _agent]
     :as auth-identity}]
-  (cond
-    ;; /statements path
-    (some-> path-info not-empty (cstr/ends-with? "statements"))
+  (let [path (or path-info "")]
     (cond
-      (#{:get :head} request-method)
-      (some? (most-permissive-statement-read-scope auth-identity))
-      (#{:put :post} request-method) ; No statement delete implemented
-      (some? (most-permissive-statement-write-scope auth-identity))
+      ;; /statements path
+      (cstr/ends-with? path "statements")
+      (cond
+        (#{:get :head} request-method)
+        (some? (most-permissive-statement-read-scope auth-identity))
+        (#{:put :post} request-method) ; No statement delete implemented
+        (some? (most-permissive-statement-write-scope auth-identity))
+        :else
+        (do (log-scope-error scopes)
+            false))
+
+      ;; activities/state
+      (and
+       (cstr/ends-with? path "activities/state")
+       (contains? scopes :scope/state))
+      true
```

Comment:
That would break `:scope/all` and `:scope/all.read` wouldn't it?

Reply:
Corrected version:
```clojure
(cstr/ends-with? path "activities/state")
(or (contains? scopes :scope/all)
    (contains? scopes :scope/all.read)
    (contains? scopes :scope/state)
```
Hmm admittedly it's now less obvious if that change would be an obvious improvement, since now we would have `:scope/all` and `:scope/all.read` in two different places.

Reply:
In _this_ one, `:scope/all.read` let's you write ;)

Reply:
Lmao I do not have my brain cells today:
```clojure
(cstr/ends-with? path "activities/state")
(or (contains? scopes :scope/all)
    (contains? scopes :scope/state)
```

Reply:
Now `:scope/all.read` doesn't let you read!

Reply:
```clojure
(cstr/ends-with? path "activities/state")
(cond
  (#{:get :head} request-method)
  (or (contains? scopes :scope/all)
      (contains? scopes :scope/all.read)
      (contains? scopes :scope/state))
  (#{:put :post :delete} request-method)
  (or (contains? scopes :scope/all)
      (contains? scopes :scope/state))
  :else
  (do (log-scope-error scopes)
      false))
```

Reply:
I think your brain cells are just fine! I think the issue is actually with the opinion encoded in
> "outer cond is predicated on the resource path, inner cond is for the actual REST op + scope check"

This construct just doesn't deal all well with `:scope/all` and `:scope/all.read`. Yes we could add additional method checking (as I'm sure you have as I write this ;), but there is a LOT of duplicated code.

Reply:
> In general I think we could refactor this function to be more declarative

Now I'm curious how _you_ would refactor this function to be more declarative. Because I do think it's something worth thinking about before we keep piling on more scopes.

Reply:
Well if we keep using `cond` I think perhaps messing with the clause order so that we can address the more permissive _scopes_ first in the short-circuit order, then do the more specific ones further down. But perhaps `cond` isn't even the best way, would have to do a little looking. I think it is a valuable refactor but it lies outside of the scope of this PR.

Reply:
Like a properly declarative way would be to check a map with routes and scopes, with handling for the overriding ones